skxray/roi.py

```diff
@@ -106,218 +106,182 @@ def rectangles(num_rois, roi_data, image_shape):
-        different steps - different step value between rings (provide
-        step value for each ring eg: 6 rings provide 5 step values)
+    angle_offset : float or array, optional
+        offset in radians from offset_angle=0 along the positive X axis
 
     Returns
     -------
-    ring_vals : array
-        edge values of each ring
-    """
-    ring_vals = []
-
-    for arg in step_r:
-        if arg < 0:
-            raise ValueError("step value for the next ring from the "
-                             "end of the previous ring has to be positive ")
-
-    if len(step_r) == 1:
-        #  when there is a same values of step between rings
-        #  the edge values of rings will be
-        ring_vals = first_r + np.r_[0, np.cumsum(np.tile([delta_r,
-                                                          float(step_r[0])],
-                                                         num_rings))][:-1]
-    else:
-        # when there is a different step values between each ring
-        #  edge values of the rings will be
-        if len(step_r) == (num_rings - 1):
-            ring_vals.append(first_r)
-            for arg in step_r:
-                ring_vals.append(ring_vals[-1] + delta_r)
-                ring_vals.append(ring_vals[-1] + float(arg))
-            ring_vals.append(ring_vals[-1] + delta_r)
-        else:
-            raise ValueError("Provide step value for each q ring ")
-
-    return ring_vals
+    label_array : array
+        Elements not inside any ROI are zero; elements inside each
+        ROI are 1, 2, 3, corresponding to the order they are specified
+        in edges and slicing
 
-
-def process_ring_edges(ring_vals):
     """
-    This function will provide edge values
-    of the each roi ring shape as (num_rings, 2)
+    agrid = core.angle_grid(center, shape)
 
-    Parameters
-    ----------
-    ring_vals : array
-        edge values of each ring
+    slicing_is_list = isinstance(slicing, collections.Iterable)
+    if slicing_is_list:
+        slicing = np.asarray(slicing) + offset_angle
+    else:
+        slicing = np.linspace(0, 360, slicing) + offset_angle
 
-    Returns
-    -------
-    ring_vals : array
-        edge values of each ring
-        shape is (num_rings, 2)
+    # the indices of the bins(angles) to which each value in input
+    #  array(angle_grid) belongs.
+    ind_grid = (np.digitize(np.ravel(agrid), slicing,
+                            right=False)).reshape(shape)
 
-    """
-    ring_vals = np.asarray(ring_vals).reshape(-1, 2)
+    label_array = np.zeros(shape, dtype=np.int64)
+    # radius grid for the image_shape
+    grid_values = core.radial_grid(center, shape)
+
+    # assign indices value according to angles then rings
+    for r in range(len(edges)):
+        vl = (edges[r][0] <= grid_values) & (grid_values
+                                                  < edges[r][1])
+        label_array[vl] = ind_grid[vl] + len(slicing)*(r)
```

Comment:
I would stash `len(sicing)` and as something like `segments_per_ring` to make it clear how this is avoiding aliasing (it took my 5min to sort out what was going on).